File: typescript/fizzbuzz.ts
function fizzbuzz(number : number) {
    var result = "";

    if (number % 3 == 0) {
        result += "Fizz";
    }

    if (number % 5 == 0) {
        result += "Buzz";
    }

    if (result === "") {
        result = number.toString();
    }

    return result;
}

var result = (function () {
    var _i = 1, _r = [];
    for (; _i <= 100; _i += 1) {
        _r.push(fizzbuzz(_i));
    }
    return _r;
}()).join("<br />");

document.write(result);

File: opal/fizzbuzz.rb
def fizzbuzz(number)
  case [number%3 == 0, number%5 == 0]
  when [true, true]
    'FizzBuzz'
  when [true, false]
    'Fizz'
  when [false, true]
    'Buzz'
  else
    number.to_s
  end
end

result = 100.times.collect { |x| fizzbuzz(x+1) }.join('<br />')

`document.write(#{result})`
